settlement/bitflyer/upload.go

```diff
@@ -20,93 +18,78 @@ var (
-			channel := payout.Channel
-			if byAttr[publisher] == nil {
-				byAttr[publisher] = make(map[string]settlement.Transaction)
+			walletProviderID := payout.WalletProviderID
+			if groupedByWalletProviderID[walletProviderID] == nil {
+				groupedByWalletProviderID[walletProviderID] = []settlement.Transaction{}
 			}
-			byAttr[publisher][channel] = payout
-			byTransferID[payout.TransferID()] = payout
-			channelsByPublisher[publisher] = append(channelsByPublisher[publisher], channel)
+			groupedByWalletProviderID[walletProviderID] = append(groupedByWalletProviderID[walletProviderID], payout)
 		}
 	}
-	for publisher, channels := range channelsByPublisher {
-		sort.Strings(channels)
-		for _, channel := range channels {
-			groupedByPublisher[publisher] = append(groupedByPublisher[publisher], byAttr[publisher][channel])
-		}
-	}
-	return groupedByPublisher
+	return groupedByWalletProviderID
 }
 
 // CategorizeResponse categorizes a response from bitflyer as pending, complete, failed, or unknown
 func CategorizeResponse(
-	limitedSettlements map[string]settlement.Transaction,
-	groupedByPublisher map[string][]settlement.Transaction,
+	batchByTransferID map[string]settlement.AggregateTransaction,
 	payout *bitflyer.WithdrawToDepositIDResponse,
-) (settlement.Transaction, string) {
-	currentTx := limitedSettlements[payout.TransferID]
+) ([]settlement.Transaction, string) {
+	currentTx := batchByTransferID[payout.TransferID]
 	key := payout.CategorizeStatus()
-	if payout.Status == "OTHER_ERROR" && !currentTx.Amount.Equal(payout.Amount) {
-		key = "complete"
-		payout.Status = "EXECUTED"
-		// use payed out amount
-		amount := payout.Amount
-		currentTx.Amount = amount
-		currentTx.Probi = altcurrency.BAT.ToProbi(amount)
-		if currentTx.Type == "contribution" {
-			currentTx.BATPlatformFee = currentTx.Probi.Div(decimal.NewFromFloat(19)).Truncate(0)
-		}
-	}
+
 	currentTx.Status = key
-	note := payout.Status
+	note := ""
 	if payout.Message != "" {
-		note = fmt.Sprintf("%s: %s", payout.Status, payout.Message)
+		note = fmt.Sprintf("%s: %s. transferID: %s", payout.Status, payout.Message, payout.TransferID)
+	} else {
+		note = fmt.Sprintf("%s transferID: %s", payout.Status, payout.TransferID)
 	}
-	// overwrite the amount
-	currentTx.Note = note
-	tmp := altcurrency.BAT
-	currentTx.AltCurrency = &tmp
-	currentTx.Currency = tmp.String()
-	return currentTx, key
+
+	for i, tx := range currentTx.Inputs {
+		// overwrite the amount
+		tx.Note = note
+		tmp := altcurrency.BAT
+		tx.AltCurrency = &tmp
+		tx.Currency = tmp.String()
+		currentTx.Inputs[i] = tx
+	}
+	return currentTx.Inputs, key
 }
 
 // CategorizeResponses categorizes the series of responses
 func CategorizeResponses(
-	limitedSettlements map[string]settlement.Transaction,
-	groupedByPublisher map[string][]settlement.Transaction,
+	batch []settlement.AggregateTransaction,
 	response *[]bitflyer.WithdrawToDepositIDResponse,
 ) map[string][]settlement.Transaction {
```

Comment:
Could the type here use an alias with a descriptive name instead of `string`? Would help to better understand at a glance what is going on. Perhaps `reponse_type` or an enum to represent pending, complete, failed, or unknown?